.github/workflows/docker_build_publish.yml

```
@@ -1,9 +1,13 @@
-name: Build and push master and develop container
+name: Build and publish develop containers
 on:
   push:
     branches:
-      - master
-      - develop
+      - "develop"
+  pull_request:
+    paths:
+      - "BALSAMIC/containers/**"
+    branches-ignore:
+      - "release_*"
```

```suggestion
    branches-ignore:
      - "release_*"
      - "master"
```
do we need to ignore `master` too?

---

Mmm good question... Could there be a situation in which the base branch for pull requests is `master`? I can't think of such a scenario, anything I'm missing?

---

That being said I don't think they should be modifying containers but as a sanity check?

---

But if we have hotfixes, the PR will have `release_vX.X.X` as a base branch.

I don't think it's possible with our model to have a master as a base branch in a PR. Our intention is only to push changes to the master branch, not to make modifications directly on it.

---

> That being said I don't think they should be modifying containers but as a sanity check?

We need to be able to track the old containers and label them with the release versions, that's the idea behind it.

---

> But if we have hotfixes, the PR will have `release_vX.X.X` as a base branch.
> 
> I don't think it's possible with our model to have a master as a base branch in a PR. Our intention is only to push changes to the master branch, not to make modifications directly on it.

True. I guess this scenario would only be possible in the case that human errors happen and we do not name the branch release_XXX?

---

But that's not the scope of this Github Action. Our goal here is to automatically push the container to docker hub from pull requests that involve changes to the containers folder, assigning a `develop` tag. We'll exclude the `release_X.X.X` branches, as they are handled by separate GitHub Actions.

The idea is to eliminate the need for manual builds during testing and we'll be able to finally use the develop containers for something.

Regarding the master branch. The only way that we'll have a PR with master as a base branch is probably if we want to open a PR to merge changes from master to develop, which is very unlikely.

Sorry I was not clear enough in the PR description, and this branch ignores/includes are quite confusing 🥵

---

Ok, I understood what you mean. My implementation is not correct because of some incorrect assumptions I made regarding `on pull request`, nice that you pointed it out 🙇 🙇 🙇

---

OK, I've fixed it. Very good call, thanks again!